olp-cpp-sdk-dataservice-read/src/StreamLayerClientImpl.cpp

```diff
@@ -280,6 +280,101 @@ client::CancellableFuture<DataResponse> StreamLayerClientImpl::GetData(
                                                       std::move(promise));
 }
 
+client::CancellationToken StreamLayerClientImpl::Poll(
+    PollResponseCallback callback) {
+  auto poll_task = [=](client::CancellationContext context) -> PollResponse {
+    std::string subscription_id;
+    std::string subscription_mode;
+    std::string x_correlation_id;
+    std::shared_ptr<client::OlpClient> client;
+
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      if (!client_context_) {
+        OLP_SDK_LOG_WARNING_F(kLogTag,
+                              "Poll: unsuccessful, not subscribed");
+
+        return client::ApiError(client::ErrorCode::PreconditionFailed,
+                                "Not subscribed", false);
+      }
+
+      subscription_id = client_context_->subscription_id;
+      subscription_mode = client_context_->subscription_mode;
+      x_correlation_id = client_context_->x_correlation_id;
+      client = client_context_->client;
+    }
+
+    OLP_SDK_LOG_INFO_F(kLogTag,
```

Comment:
```suggestion
    OLP_SDK_LOG_DEBUG_F(kLogTag,
```
The info level is a little too much, isn't it? Do we wish to be informed about every started poll request?

Reply:
All other similar log messages for other methods in this class and for other Layer Clients Impl classes has Info level. So to be consistent we should leave this message as is or change log level for all the similar messages in all similar classes.

Reply:
It makes sense. We need apparently a separate ticket to review and correct log levels across the SDK. I'll check if we have one already and create a new if not.

Reply:
I have changed log level in this file for all functions.